Create HandcalcsDemo.jl

co1emi11er2 opened the pull request:
Hi @fonsp! I was able to get a rough Handcalcs.jl demo notebook made. I added it to the math folder, but I am fine with wherever you want to put it! Thanks again for this notebook feature. I am honored to potentially have a featured notebook!

Let me know what you think of the notebook!

co1emi11er2 commented:
Hi @fonsp! Did you have time to take a look at this? Thanks again!

co1emi11er2 commented:
I added a preview image and an author image. Is there anything else I need to do to get this merged? Thanks!

fonsp commented:
Thanks for the featured notebook, it looks really good!

Sorry for the delay, I don't know how it slipped my inbox :(

I made a couple of small formatting changes, the rest looks really good!

For the project demo, can you add an introduction? Right now the introduction is: "Determine the LRFD flexural design strength for a W10x12 beam with an unbraced length of 10 ft." Maybe you could add one more sentence before this, to explain to a non-engineer what is about to happen. Links to other materials would also be really nice, e.g. "Determine Flexural Capacity based on Chapter F, Section 3" makes me wonder what book you are using.

And maybe `calc_Ix` can just be defined inside the notebook, inside of importing it. Unless you want to demonstrate that imported functions can be unrolled.

Thanks again!

fonsp commented:
Could you try to use https://github.com/JuliaPluto/PlutoPlotly.jl for the plot instead of Plots.jl? Since it's only one plot, loading the package Plots.jl is a bit heavy. If not, thats okay!

co1emi11er2 commented:
> Thanks for the featured notebook, it looks really good!
> 
> Sorry for the delay, I don't know how it slipped my inbox :(
> 
> I made a couple of small formatting changes, the rest looks really good!

Thanks! No Worries.

> For the project demo, can you add an introduction? 

Yes. Good idea. I added a note to the reader and a link to the pdf that I reference in the calc for users to be able to follow along.

> And maybe `calc_Ix` can just be defined inside the notebook, inside of importing it. Unless you want to demonstrate that imported functions can be unrolled.

Unfortunately, functions defined in pluto/jupyter or even the same file can not be found by the `code_expr` macro from the CodeTracking.jl package (which Handcalcs relies on for this). I would really like to fix that but that package is too complex for me at the moment. For now, functions need to be defined in a package. There is a work around with `includet` from Revise.jl that allows you to use functions from another file, but I haven't figured out how to get that working in pluto. 

I added a note further clarifying that limitation.

> Could you try to use https://github.com/JuliaPluto/PlutoPlotly.jl for the plot instead of Plots.jl? Since it's only one plot, loading the package Plots.jl is a bit heavy. If not, thats okay!

Okay sure! I updated to only use PlutoPlotly.jl.

> Thanks again!

Thank you as well for letting me put out a featured notebook! I am honored!

fonsp commented:
Really nice, thank you!!


Are you interested in submitting another featured notebook? :)


![](https://media4.giphy.com/media/v1.Y2lkPTc5MGI3NjExbTJwOHpjZm81dTVtMzk0NjIwb3A3bzlyYTU1cWc3cGZlanE3cnVpaSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/kLe9PygrSqB4oO4tYV/giphy.gif)

co1emi11er2 commented:
I created a new pull request #79. Hopefully that fixes the reactivity issues. Sorry for the inconvenience!
